### PR TITLE
[#1086] [Doc] Simplify the Gluten code and add the doc

### PR DESCRIPTION
**Pull request by summaryzb**
### What changes were proposed in this pull request?
1. Simplify  the code when integrate with gluten
2. Add gluten uniffle doc

### Why are the changes needed?
Integrate with gluten before gluten and uniffle release

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test since both projects are not released, after release, I'll add some tests in both projects

### CODE REVIEW

**Comment by summaryzb** [comment]
@leixm @yl09099 @jerqi 
CommitID `0da1dd1164437a4cd` in branch-0.8 seems like incomplete， it has make the unit test fail

**Comment by summaryzb** [comment]
Maybe we should import all the related commits in [Dynamic shuffle server assignment ](https://github.com/apache/incubator-uniffle/issues/825),  or just remove `0da1dd1164437a4cd`

**Comment by summaryzb** [comment]
@advancedxy @xianjingfeng PTAL

**Comment by jerqi** [comment]
branch 0.8 has reverted 0da1dd1164437a4cd. Could you rebase this branch?

File: client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java [hunk 1]
```diff
@@ -162,9 +162,10 @@ public WriteBufferManager(
   }
 
   /** add serialized columnar data directly when integrate with gluten */
-  public List<ShuffleBlockInfo> addPartitionData(int partitionId, byte[] serializedData) {
+  public List<ShuffleBlockInfo> addPartitionData(
+      int partitionId, byte[] serializedData, int serializedDataLength) {
```
Comment (summaryzb) [inline client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java:166]: Currently gluten reuse this `serializedData` byte array, only if `serializedDataLength` may be shorter than length of `serializedData`

**Comment by xianjingfeng** [comment]
I am publishing 0.8.0-rc3. Is this pr necessary.

**Comment by summaryzb** [comment]
> I am publishing 0.8.0-rc3. Is this pr necessary. @summaryzb

@xianjingfeng Yes, thanks

File: client-spark/common/src/test/java/org/apache/spark/shuffle/writer/WriteBufferManagerTest.java [hunk 2]
```diff
@@ -211,10 +211,10 @@ public void addPartitionDataTest() {
     WriteBufferManager spyManager = spy(wbm);
     doReturn(512L).when(spyManager).acquireMemory(anyLong());
 
-    List<ShuffleBlockInfo> shuffleBlockInfos = spyManager.addPartitionData(0, new byte[64]);
+    List<ShuffleBlockInfo> shuffleBlockInfos = spyManager.addPartitionData(0, new byte[64], 64);
     assertEquals(1, spyManager.getBuffers().size());
     assertEquals(0, shuffleBlockInfos.size());
-    shuffleBlockInfos = spyManager.addPartitionData(0, new byte[64]);
+    shuffleBlockInfos = spyManager.addPartitionData(0, new byte[64], 64);
```
[the remark this answers is not in the document]
Comment (summaryzb) [reply client-spark/common/src/test/java/org/apache/spark/shuffle/writer/WriteBufferManagerTest.java:217]: Ok

**Comment by summaryzb** [comment]
> partitionId, byte[] serializedData, int serializedDataLength) {

**Comment by summaryzb** [comment]
> This PR will be cherry-picked into master, right?
Yes

File: client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java [hunk 3]
```diff
@@ -86,7 +86,7 @@ public class RssShuffleWriter<K, V, C> extends ShuffleWriter<K, V> {
   private final Map<Integer, List<ShuffleServerInfo>> partitionToServers;
   private final Set<ShuffleServerInfo> shuffleServersForData;
   private final long[] partitionLengths;
-  private final boolean isMemoryShuffleEnabled;
+  protected final boolean isMemoryShuffleEnabled;
```
[the remark this answers is not in the document]
Comment (summaryzb) [reply client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java:89]: Protected field and method can be accessed in the same scope,  and the final field can not be changed except in the constructor.

File: client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java (hunk 1 above)
Comment (summaryzb) [reply client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java:166]: Ok, remove this and change the title

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Comment by xianjingfeng** [comment]
@summaryzb Could you cherry-pick this pr into master?

**Pull request merged**